File: Java/Dagger/atm/src/main/java/ro/razvan/java/dagger/atm/CommandRouterFactory.java
package ro.razvan.java.dagger.atm;

import dagger.Component;

import javax.inject.Singleton;

@Component(modules = {
        LoginCommandModule.class,
        HelloWorldCommandModule.class,
        DepositCommandModule.class,
        SystemOutModule.class
})
@Singleton
public interface CommandRouterFactory {

    CommandRouter router();

    static CommandRouterFactory create() {
        return DaggerCommandRouterFactory.create();
    }
}

File: Kotlin/Dagger/atm/src/main/kotlin/ro/razvan/kotlin/dagger/atm/CommandRouterFactory.kt
package ro.razvan.kotlin.dagger.atm

import dagger.Component
import javax.inject.Singleton

@Component(
    modules = [
        LoginCommandModule::class,
        HelloWorldCommandModule::class,
        SystemOutModule::class,
        DepositCommandModule::class
    ]
)
@Singleton
interface CommandRouterFactory {
    fun router(): CommandRouter

    companion object {
        fun create(): CommandRouterFactory =
            DaggerCommandRouterFactory.create()
    }
}